Find a functionally equivalent code:javax.obex.ClientOperation.startProcessing:COMMENT
Method Modifier: private     
Comment:/**
 * This method starts the processing thread results. It will send the
 * initial request. If the response takes more then one packet, a thread
 * will be started to handle additional requests
 * @throws IOException if an IO error occurs
 */

@@ -5,18 +5,26 @@
     boolean more = true;
     if (mGetOperation) {
         if (!mOperationDone) {
-            mReplyHeader.responseCode = ResponseCodes.OBEX_HTTP_CONTINUE;
-            while ((more) && (mReplyHeader.responseCode == ResponseCodes.OBEX_HTTP_CONTINUE)) {
-                more = sendRequest(0x03);
-            }
-            if (mReplyHeader.responseCode == ResponseCodes.OBEX_HTTP_CONTINUE) {
-                mParent.sendRequest(0x83, null, mReplyHeader, mPrivateInput);
-            }
-            if (mReplyHeader.responseCode != ResponseCodes.OBEX_HTTP_CONTINUE) {
+            if (!mGetFinalFlag) {
+                mReplyHeader.responseCode = ResponseCodes.OBEX_HTTP_CONTINUE;
+                while ((more) && (mReplyHeader.responseCode == ResponseCodes.OBEX_HTTP_CONTINUE)) {
+                    more = sendRequest(0x03);
+                }
+                if (mReplyHeader.responseCode == ResponseCodes.OBEX_HTTP_CONTINUE) {
+                    mParent.sendRequest(0x83, null, mReplyHeader, mPrivateInput);
+                }
+                if (mReplyHeader.responseCode != ResponseCodes.OBEX_HTTP_CONTINUE) {
+                    mOperationDone = true;
+                }
+            } else {
+                more = sendRequest(0x83);
+                if (more) {
+                    throw new IOException("FINAL_GET forced but data did not fit into single packet!");
+                }
                 mOperationDone = true;
             }
         }

